nautobot/core/testing/views.py

```diff
@@ -1481,6 +1502,38 @@ class DeviceComponentTemplateViewTestCase(
 
         maxDiff = None
 
+        def _edit_object_test_setup(self):
+            test_instance = self._get_queryset().first()
+            self.update_data = {
+                "name": test_instance.name,
+                "device_type": test_instance.device_type.pk,
+                "label": "new test label",
+                "description": "new test description",
+            }
+
+        @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+        def test_edit_object_with_constrained_permission(self):
+            # Overload this test so that only the label and description fields are changed
+            self._edit_object_test_setup()
+            super().test_edit_object_with_constrained_permission()
+
+        @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+        def test_edit_object_with_permission(self):
+            # Overload this test so that only the label and description fields are changed
+            self._edit_object_test_setup()
+            super().test_edit_object_with_permission()
+
+    class ModularDeviceComponentTemplateViewTestCase(DeviceComponentTemplateViewTestCase):
+        def _edit_object_test_setup(self):
+            test_instance = self._get_queryset().first()
+            self.update_data = {
+                "name": test_instance.name,
+                "device_type": getattr(getattr(test_instance, "device_type", None), "pk", None),
+                "module_type": getattr(getattr(test_instance, "module_type", None), "pk", None),
+                "label": "new test label",
+                "description": "new test description",
+            }
```

Comment:
With the creation of the component factories, this test was setting statically defined data on randomly generated instances. That doesn't work for cases like `FrontPort` where the `RearPort` has to belong to the same device/module. We should update this test to work like the API tests where we can supply specific field values to change instead of having to provide the entire set of form data.